nlp_primitives/number_of_hashtags.py

```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8 -*-
+import numpy as np
+from featuretools.primitives.base import TransformPrimitive
+from woodwork.column_schema import ColumnSchema
+from woodwork.logical_types import Double, NaturalLanguage
+
+
+class NumberOfHashtags(TransformPrimitive):
+    """Determines the number of hashtags in a string.
+
+    Description:
+        Given list of strings, determine the number of hashtags
+        in each string. A hashtag is defined as any word starting
+        with a '#' sign followed by a sequence of alphanumeric characters
+        or underscores.
+
+        If a string is missing, return `NaN`.
+
+    Examples:
+        >>> x = ['#regular#expression', 'this is a string', '###__regular#1and_0#expression']
+        >>> number_of_hashtags = NumberOfHashtags()
+        >>> number_of_hashtags(x).tolist()
+        [2.0, 0.0, 3.0]
+    """
+
+    name = "number_of_hashtags"
+    input_types = [ColumnSchema(logical_type=NaturalLanguage)]
+    return_type = ColumnSchema(logical_type=Double, semantic_tags={"numeric"})
+    default_value = 0
+
+    def get_function(self):
+        pattern = r"(#[A-Za-z0-9|\_]+)"
+
+        def number_of_hashtags(x):
```

Comment:
We have a CountString primitive that has similar functionality
- https://github.com/alteryx/nlp_primitives/blob/main/nlp_primitives/count_string.py
We leverage this primitive in other primitives, like WhiteSpaceCount
- https://github.com/alteryx/nlp_primitives/blob/main/nlp_primitives/whitespace_count.py#L6

Should we take the same approach here? Is this a faster implementation (that isn't possible with CountString)?

Reply:
This primitive would fit well with CountString. One thing is that using CountString would return a np.ndarray but I noticed most of the other primitives return a pd.Series.
I think there are a few other primitives that could be rewritten using CountString as well (maybe more?):

TitleWordCount
UpperCaseCount
PunctuationCount

Reply:
I believe Featuretools can accept either return type. 

Let's create an issue to re-write the other primitives.

Reply:
Sounds good!